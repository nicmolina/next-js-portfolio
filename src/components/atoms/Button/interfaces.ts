import { ReactNode } from 'react'

export type ButtonVariant = 'primary' | 'secondary'

export interface IButtonProps {
  variant: ButtonVariant
  icon?: ReactNode
  onClick?: (event?: any) => void
  type?: 'button' | 'submit' | 'reset'
  label?: string
  children?: ReactNode
  isLoading?: boolean
  disabled?: boolean
  size?: 'small' | 'medium' | 'large'
}
