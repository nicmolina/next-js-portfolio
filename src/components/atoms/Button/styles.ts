import styled from 'styled-components'
import { IButtonProps } from './interfaces'

export const Button = styled.button<IButtonProps>`
  display: flex;
  width: 100%;
  height: 100;
  background-color: ${(props) => props.variant === 'primary'? }
`
