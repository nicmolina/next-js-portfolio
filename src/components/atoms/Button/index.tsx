import { FC } from 'react'
import * as S from './styles'
import { IButtonProps } from './interfaces'

const Button: FC<IButtonProps> = ({
  icon,
  label,
  onClick,
  variant,
  type,
  children
}) => {
  return (
    <S.Button onClick={onClick} type={type} variant={variant}>
      {label}
      {icon}
      {children}
    </S.Button>
  )
}

export default Button
