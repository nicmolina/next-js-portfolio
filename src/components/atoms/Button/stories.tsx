import Button from '.'

export default {
  title: 'Button',
  component: Button
}

export const Basic = () => <Button variant={'primary'} />
