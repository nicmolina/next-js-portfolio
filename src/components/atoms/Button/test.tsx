import { render, screen } from '@testing-library/react'
import Button from '.'

describe('<Button/>', () => {
  it('should render a primary variant button component, match snapshot and byrole', () => {
    const { container } = render(<Button variant="primary" />)

    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should render a secondary variant button component, match snapshot and byrole', () => {
    const { container } = render(<Button variant="secondary" />)

    expect(screen.getByRole('button')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
