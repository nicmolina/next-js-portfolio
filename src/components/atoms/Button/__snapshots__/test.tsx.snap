// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> should render a button component, match snapshot and byrole 1`] = `
<button
  class="styles__Button-dzveYr fqoaUl"
/>
`;
